packages/keychain/src/components/connect/RegisterSession.tsx

```diff
@@ -28,17 +35,79 @@ export function RegisterSession({
   const [transactions, setTransactions] = useState<Call[] | undefined>(
     undefined,
   );
+  const [isEditable, setIsEditable] = useState(false);
+  const [policyState, setPolicyState] = useState<ParsedSessionPolicies>(() => {
```

Comment:
can we move most of this into the context / a hook? so the implementation isn't duplicated between here and create session?